python/tutorials/06-fused-attention.py

```diff
@@ -609,6 +609,7 @@ def bench_flash_attention(BATCH, H, N_CTX, HEAD_DIM, causal, mode, provider, dev
         if mode == "fwd" and "fp8" in provider:
             q = q.to(torch.float8_e5m2)
             k = k.to(torch.float8_e5m2)
+            v = v.permute(0, 1, 3, 2).contiguous()
```

Comment:
@ThomasRaoux I am trying to fix the FP8 flash attention here. From the kernel implementation, the block pointer is using strides of (stride_vk, stride_vn), without the fix, stride_vk should be 1, and stride_vn is HEAD_DIM. The block shape for v_block_ptr is (BLOCK_N, HEAD_DIM), it seems incorrect to have stride of HEAD_DIM for the last dimension, i.e we will have [0:HEAD_DIM)* HEAD_DIM for the offset.

The fix is to let v have shape of [B, H, N_CTX, HEAD_DIM] similar to non-fp8 case and the strides will be [x, y, 1, N_CTX], prior to the fix, the shape is [B, H, HEAD_DIM, N_CTX] with strides of [x, y, 1, HEAD_DIM]. In comparison, non-fp8 has shape of [B, H, N_CTX, HEAD_DIM]  with strides of [x, y, HEAD_DIM, 1].

It is possible that there is no issue.

Reply:
I think this is right.